File: cpp/07/pe7.cpp
#include <iostream>
#include <stdlib.h>
#include <math.h>

bool is_prime(unsigned int num);

int main(){
    system("CLS");
        int primes = 1, HIGHEST_PRIME = 10001;
        unsigned long int num = 1;
        while(primes < HIGHEST_PRIME){
            if(is_prime(num += 2)) primes++;
        }
        std::cout<<num<<std::endl;
    return 0;
}

bool is_prime(unsigned int num){
    if(num == 2) return true;
    else if(!(num%2)) return false;
    else{
        for(unsigned int i=3; i<=(sqrt(num + 1)); i+=2){
            if(!(num%i)) return false;
        }
        return true;
    }
}

File: python/pe7.py
from math import sqrt
from os import system
system('cls')

def is_prime(num):
    if num == 2: return True
    elif not(num%2): return False
    else:
        for i in range(3, int(sqrt(num)) + 1, 2):
            if(not(num%i)): return False
        return True

if __name__ == '__main__':
    HIGHEST_PRIME = 10001
    primes = 1
    num = 1
    while(primes < HIGHEST_PRIME):
        num += 2
        if(is_prime(num)): primes += 1
    print(num)
